Userland/Services/NotificationServer/NotificationWindow.cpp

```diff
@@ -60,14 +62,28 @@ void update_notification_window_locations()
 
 NotificationWindow::NotificationWindow(const String& text, const String& title, const Gfx::ShareableBitmap& icon)
 {
-    s_windows.append(this);
+    while (true) {
+        m_window_id = get_random<i64>();
+        if (m_window_id < 0) {
+            if (m_window_id == (i64)0xFFFFFFFFFFFFFFFF) {
+                m_window_id = 0;
+            } else {
+                m_window_id = -m_window_id;
+            }
+        }
+        if (!s_windows.contains(m_window_id)) {
+            break;
+        }
+    }
+    s_windows.set(m_window_id, this);
 
     set_window_type(GUI::WindowType::Notification);
     set_resizable(false);
     set_minimizable(false);
 
     Gfx::IntRect lowest_notification_rect_on_screen;
-    for (auto& window : s_windows) {
+    for (auto& window_ent : s_windows) {
```

Comment:
Nit: `entry`, to avoid contusion with `entity`? Or maybe I just read LibC for too long ^^'

Reply:
confusion* – Oh the irony of making a typo in a comment that's complaining about spelling. XD